pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+// this was copied from where else and edited to fit our objects
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+/* Begin Create Definition */
+
+type brokerCreateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+	kapi.NameGenerator  // GenerateName method for CreateStrategy
+}
+
+// implements RESTCreateStrategy interface
+var createStrategy = brokerCreateStrategy{
+	// embeds to pull in existing code behavior from upstream
+
+	// this has an interesting NOTE on it. Not sure if it applies to us.
```

Comment:
In reverse order because that's how slackarchive rolls. Start at the bottom and read upwards.
```
deads2k Jan 9, 2017 11:21 
@mbauer I really don't know that I'd want to be first. If you wait about 2 weeks, we'll be pushing hard on https://github.com/kubernetes/kubernetes/pull/39475 and will probably uncover issues
Don't panic. The diff is quite large, but its all generated change. The first few commits are where are all the action is. I built a script to find the fanout from k8s.io/kubernetes/pkg/apimachin...
@mbauer well, the idea is that you would instantiate a different scheme and codec factory for your API server, then you would use those to wire up your encoders and decoders for your entire API servers

mbauer Jan 9, 2017 11:16 
I don't mind forging new ground. We can always fall back. Any tips on schemeing?

deads2k Jan 9, 2017 11:16 
@mbauer Ok is relative. Right now it's the only example we've got. It's more correct to create a different scheme, but I can't actually be sure the code will function properly after you've created your own scheme. It *should*, but that doesn't mean it does

mbauer Jan 9, 2017 11:07 
(everywhere, not just mine)
That's what I've been doing. I see apis referencing that var, so is it okay for me to do so? It's being used for it's implementation of ObjectTyper.

deads2k Jan 9, 2017 11:05 
@mbauer Yeah. The tooling is happier with /apis and if you ever need more than one, you'll be happier too

mbauer Jan 9, 2017 11:04 
Is it in reference to the whole of pkg/api? That is, the idea of pkg/api is the old way and pkg/apis is the new way?

lavalamp Jan 9, 2017 11:01 
@mbauer That comment means use a different api group as your template if you're trying to make a new API group.

mbauer Jan 9, 2017 08:45 
Is this comment NOTE on the var Scheme? https://github.com/kubernetes/kubernetes/blame/master/pkg/api/register.go#L26-L31 I'm writing an api and cribbing from another api which references that Scheme object to implement it's Strategy. Not sure what copying this file means or what copy the extensions group instead is supposed to mean.
```